workflow/react-project/src/components/Header/index.jsx

```diff
@@ -1,18 +1,25 @@
 import { useState } from 'react';
-import { useDispatch } from 'react-redux';
+import { useDispatch, useSelector } from 'react-redux';
 
 import { faMartiniGlassCitrus } from '@fortawesome/free-solid-svg-icons';
+import { faMagnifyingGlass } from '@fortawesome/free-solid-svg-icons';
+import { faStar } from '@fortawesome/free-regular-svg-icons';
+import { faHouseUser } from '@fortawesome/free-solid-svg-icons';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
-import { changeModalKey } from '@store';
 
 import Button from '@common/Button';
-import Modal from '@common/Modal';
+import SignUp from '../Modals/Authentication/SignUp';
+import SignIn from '../Modals/Authentication/SignIn';
 
+import { changeModalKey } from '@store/appConfigurations/action';
+import { selectAuthenticationToken } from '@store/authentication/selectors';
 import './style.less';
 
 const Header = () => {
   const dispatch = useDispatch();
+  const authenticationToken = useSelector(selectAuthenticationToken);
   const [isAuthenticationModalOpened, setAuthenticationModalOpened] = useState(false);
+  const [isSignInModalOpened, setSignInModalOpened] = useState(true);
 
   const handleModalStateChange = modalKey => {
```

Comment:
handleModalStateChange - this function don't change state of modal it's open/closed. Better - `toggleModal`
as well it's better rewrite a little: 
```
const willOpened = !isAuthenticationModalOpened;
setAuthenticationModalOpened(willOpened);
 dispatch(changeModalKey(willOpened ? modalKey : null));
```